File: app/src/main/java/com/keelim/practice6/game_mode/Game5_StartVideo.java
package com.keelim.practice6.game_mode;

import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;
import android.widget.VideoView;

import androidx.appcompat.app.AppCompatActivity;

import com.keelim.practice6.R;

public class Game5_StartVideo extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.game5_activity_video_start);
        VideoView vv = (VideoView) findViewById(R.id.vv);

        // http://www ~~~ possible
        String uriPath = "android.resource://" + getPackageName() + "/" + R.raw.game3start;
        Uri uri = Uri.parse(uriPath);
        vv.setVideoURI(uri);
        vv.requestFocus();
        vv.start();


        // Event
        vv.setOnInfoListener(new MediaPlayer.OnInfoListener() {
            @Override
            public boolean onInfo(MediaPlayer mp, int what, int extra) {
                switch (what) {
                    case MediaPlayer.MEDIA_ERROR_TIMED_OUT:
                        Toast.makeText(getApplicationContext(), "MEDIA_ERROR_TIMED_OUT", Toast.LENGTH_LONG).show();
                        break;

                    case MediaPlayer.MEDIA_INFO_BUFFERING_START:
                        // Progress Diaglog 출력(Print)
                        Toast.makeText(getApplicationContext(), "MEDIA_INFO_BUFFERING_START", Toast.LENGTH_LONG).show();
                        break;

                    case MediaPlayer.MEDIA_INFO_BUFFERING_END:
                        // Progress Dialog 삭제(Delete)
                        Toast.makeText(getApplicationContext(), "MEDIA_INFO_BUFFERING_END", Toast.LENGTH_LONG).show();
                        break;
                }
                return false;
            }
        });


        // 마우스 클릭
        Button SkipButton = findViewById(R.id.SkipButton_Id);
        SkipButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });


    }
}

File: app/src/main/java/com/keelim/practice6/view/Game5_StartVideo.kt
package com.keelim.practice6.view

import android.media.MediaPlayer
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import android.widget.VideoView
import androidx.appcompat.app.AppCompatActivity
import com.keelim.practice6.R

class Game5_StartVideo : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.game5_activity_video_start)
        val vv = findViewById<View>(R.id.vv) as VideoView
        // http://www ~~~ possible
        val uriPath =
            "android.resource://" + packageName + "/" + R.raw.game3start
        val uri = Uri.parse(uriPath)
        vv.setVideoURI(uri)
        vv.requestFocus()
        vv.start()
        // Event
        vv.setOnInfoListener { mp, what, extra ->
            when (what) {
                MediaPlayer.MEDIA_ERROR_TIMED_OUT -> Toast.makeText(
                    applicationContext,
                    "MEDIA_ERROR_TIMED_OUT",
                    Toast.LENGTH_LONG
                ).show()
                MediaPlayer.MEDIA_INFO_BUFFERING_START ->  // Progress Diaglog 출력(Print)
                    Toast.makeText(
                        applicationContext,
                        "MEDIA_INFO_BUFFERING_START",
                        Toast.LENGTH_LONG
                    ).show()
                MediaPlayer.MEDIA_INFO_BUFFERING_END ->  // Progress Dialog 삭제(Delete)
                    Toast.makeText(
                        applicationContext,
                        "MEDIA_INFO_BUFFERING_END",
                        Toast.LENGTH_LONG
                    ).show()
            }
            false
        }
        // 마우스 클릭
        val SkipButton =
            findViewById<Button>(R.id.SkipButton_Id)
        SkipButton.setOnClickListener { finish() }
    }
}